src/main/java/club/mher/compass/listener/GameListenerBW2023.java

```diff
@@ -0,0 +1,90 @@
+package club.mher.compass.listener;
+
+import club.mher.compass.Compass;
+import club.mher.compass.data.BW1058MainConfig;
+import club.mher.compass.util.NBTItem;
+import com.tomkeuper.bedwars.api.arena.GameState;
+import com.tomkeuper.bedwars.api.arena.IArena;
+import com.tomkeuper.bedwars.api.arena.team.ITeam;
+import com.tomkeuper.bedwars.api.events.gameplay.GameStateChangeEvent;
+import com.tomkeuper.bedwars.api.events.player.PlayerKillEvent;
+import com.tomkeuper.bedwars.api.events.player.PlayerLeaveArenaEvent;
+import com.tomkeuper.bedwars.api.events.player.PlayerReSpawnEvent;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.ItemSpawnEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.UUID;
+
+public class GameListenerBW2023 implements Listener {
+
+    @EventHandler
+    public void onServerQuit(PlayerQuitEvent e) {
+        Player player = e.getPlayer();
+        UUID uuid = player.getUniqueId();
+        if (!Compass.getBedWars().isPlaying(player)) return;
+        Object arena = Compass.getBedWars().getArenaByPlayer(player);
+        if (!Compass.isTracking(arena, uuid)) return;
+        Compass.removeTrackingTeam(arena, uuid);
+    }
+
+    @EventHandler
+    public void onLeave(PlayerLeaveArenaEvent e) {
+        IArena arena = e.getArena();
+        UUID uuid = e.getPlayer().getUniqueId();
+        if (Compass.isTracking(arena, uuid)) Compass.removeTrackingTeam(arena, uuid);
+    }
+
+    @EventHandler
+    public void onKill(PlayerDeathEvent e) {
+        Player player = e.getEntity();
+        if (!Compass.getBedWars().isPlaying(player)) return;
+        NBTItem nbti = new NBTItem(Compass.getMainConfig().getItem(player, BW1058MainConfig.COMPASS_ITEM, true, "compass-item"));
```

Comment:
On this class, the NBTItems use the items from bw1058 config and not bw2023

Reply:
Solved! Good spotting!